### PR TITLE
Frances

### LINKED ISSUE #3: As a player, I want to add new QR codes to my account. (US 01.01.01)
As a player, I want to add new QR codes to my account.

## Acceptance Criteria
- [ ] Users should be able to user their camera to take images containing QR codes
- [ ] As a user is taking an image, the QR code should be detected
- [ ] Once a user takes an image of the QR code it should be stored on their profile and in the database

Links that may be helpful in the future:
https://developers.google.com/ml-kit/vision/barcode-scanning/android#java

### LINKED ISSUE #8: As a player, I want to remove QR codes from my account. (US 01.03.01)
As a player, I want to remove QR codes from my account.

## Acceptance Criteria
- [ ] There should be a list of QR codes that the player scanned that is accessible
- [ ] It should be possible to remove a QR code from a user's account & score
- [ ] Users should be prompted to confirm they want to remove a QR code

### PR DESCRIPTION
**Pull request by IEfrances**
I have implemented the buttons to add and delete qrcodes. it should work fine.

### CODE REVIEW

File: app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java [hunk 1]
```diff
@@ -39,13 +44,16 @@ public Player(UUID deviceId, String username, String phoneNo, String email) {
    * @param username username of the player
    * @param phoneNo phone number of the player
    * @param email email of the player
+   * @param qrCodeHashes qrcode hashes associated with player
    */
-  public Player(String documentId, UUID deviceId, String username, String phoneNo, String email) {
+  public Player(String documentId, UUID deviceId, String username, String phoneNo, String email, ArrayList<String> qrCodeHashes) {
```
Comment (jmmabanta) [inline app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java:49]: Can you rename qrCodeHashes to `scannedQRHashes` so it's more descriptive about what it represents?

File: build.gradle [hunk 2]
```diff
@@ -11,6 +11,6 @@ buildscript {
 }
 
 plugins {
-    id 'com.android.application' version '7.4.1' apply false
-    id 'com.android.library' version '7.4.1' apply false
+    id 'com.android.application' version '7.3.1' apply false
+    id 'com.android.library' version '7.3.1' apply false
 }
```
Comment (jmmabanta) [inline build.gradle:16]: Any reason why this was downgraded?

**Review by jmmabanta — CHANGES_REQUESTED** [review]
Made two small comments, and do the following:
- Change TestPlayerModel.java so that an empty arraylist is passed for the scannedQRList since you modified the player constructor. This should make the test case pass.
- Run `./gradlew spotlessApply` to format your code

I'll also wait for #40 to be merged first before this one. This is because I made some improvements to how the player can be accessed that I would like to apply to this branch.

File: app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java [hunk 3]
```diff
@@ -151,4 +159,13 @@ public void setEmail(String email) {
...
   }
+
+  // Get all the QR code hashes associated with this player
+  public ArrayList<String> getQRCodeHashes() {
+    return qrCodeHashes;
+  }
+
+  public void setQRCodeHashes(List<String> qrcodeHashes) {
+    this.qrCodeHashes = new ArrayList<>(qrcodeHashes);
+  }
```
Comment (jmmabanta) [inline app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java:170]: Can these also have the docstring format so they are logged correctly in the Javadoc?

File: build.gradle (hunk 2 above)
Comment (FrancesIgwe) [reply build.gradle:16]: i get build fail when i run the 7.4.1 version. i need to change that

File: app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java (hunk 1 above)
Comment (FrancesIgwe) [reply app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java:49]: ok

File: app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java (hunk 3 above)
Comment (FrancesIgwe) [reply app/src/main/java/com/cmput301w23t09/qrhunter/player/Player.java:170]: ok

File: build.gradle (hunk 2 above)
Comment (jmmabanta) [reply build.gradle:16]: Ohh ok

**Comment by jmmabanta** [comment]
#40 has been merged
I'll fix these merge conflicts since they had to do with my changes

**Pull request merged**